google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPoolAsyncTransactionManager.java

```diff
@@ -167,9 +171,12 @@ public void onError(Throwable t) {
   public ApiFuture<Timestamp> commitAsync() {
     synchronized (lock) {
       Preconditions.checkState(
-          txnState == TransactionState.STARTED,
+          txnState == TransactionState.STARTED || txnState == TransactionState.ABORTED,
           "commit can only be invoked if the transaction is in progress. Current state: "
               + txnState);
+      if (txnState == TransactionState.ABORTED) {
```

Comment:
This makes sure that the `AsyncTransactionManager` will return an `ApiFuture` with exception `Aborted` if the transaction has already been aborted by an asynchronous statement that has not yet been checked by the application, e.g. (simplified):
```java
txn.updateAsync(...); // This aborts the transaction, but the status of the returned ApiFuture is not checked by the application.
...
txn.commitAsync();
```